### PR TITLE
refactor(localdebug): support migrate v1 scaffold local debug settings

### PR DESCRIPTION
**Pull request by XiaofuHuang**
Migrate v1 should use the local debug settings scaffolding logic in the solution.

### CODE REVIEW

**Comment by swatDong** [comment]
Any unit test for migrate v1?

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/telemetry.ts [hunk 1]
```diff
@@ -22,17 +23,17 @@ enum TelemetryPropertyValue {
...
-  setupLocalDebugSettings = "setupLocalDebugSettings",
-  configLocalDebugSettings = "configLocalDebugSettings",
+  scaffoldLocalDebugSettings = "scaffold-local-debug-settings",
+  setupLocalDebugSettings = "setup-local-debug-settings",
+  configLocalDebugSettings = "config-local-debug-settings",
 }
 
 export class TelemetryUtils {
-  static ctx: v2.Context;
+  static telemetryReporter: TelemetryReporter;
```
Comment (swatDong) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/util/telemetry.ts:32]: `TelemetryReporter | undefined`?

File: packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts [hunk 2]
```diff
@@ -135,15 +180,15 @@ export async function scaffoldLocalDebugSettings(
...
+        await scaffoldLocalSettingsJson(projectSetting, inputs, cryptoProvider, localSettings);
 
         // add 'npm install' scripts into root package.json
         const packageJsonPath = inputs.projectPath;
         let packageJson: any = undefined;
         try {
           packageJson = await PackageJson.load(packageJsonPath);
         } catch (error) {
-          ctx.logProvider?.error(`Cannot load package.json from ${inputs.projectPath}. ${error}`);
+          logProvider?.error(`Cannot load package.json from ${inputs.projectPath}. ${error}`);
```
Comment (swatDong) [inline packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts:191]: Will `logProvider` be undefined?

File: packages/fx-core/src/plugins/solution/fx-solution/debug/util/telemetry.ts (hunk 1 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/solution/fx-solution/debug/util/telemetry.ts:32]: Thanks, forget to change v1 optional reporter and provider to v2 required reporter.

**Comment by XiaofuHuang** [comment]
> Any unit test for migrate v1?
The origin UT only validate projectSettings.json
Update the unit test to validate the local debug scaffolding result.

File: packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts (hunk 2 above)
Comment (XiaofuHuang) [reply packages/fx-core/src/plugins/solution/fx-solution/debug/scaffolding.ts]: Updated, validate the v1 context before scaffolding.

**Review by kuojianlu — APPROVED** [review]

**Review by swatDong — APPROVED** [review]

**Review by jayzhang — APPROVED** [review]

**Review by 1yefuwang1 — APPROVED** [review]

**Pull request merged**